app/src/js/server/bdd_type_transformers.ts

```diff
@@ -0,0 +1,57 @@
+import { ShapeTypeName } from '../common/types'
+import * as types from '../functional/types'
+import * as bdd from './bdd_types'
+
+/**
+ * Transform internal rect to export 2d box
+ * @param shape
+ */
+export function transformBox2D (shape: types.ShapeType): bdd.Box2DType {
```

Comment:
Why do we need these transformations, since the two types have the same properties?

Reply:
Those transformations are necessary to remove the additional fields in the internal shapes.